src/components/DataWrapper.tsx

```diff
@@ -0,0 +1,32 @@
+import { QueryErrorResetBoundary } from '@tanstack/react-query';
+import { ErrorBoundary } from 'react-error-boundary';
+import { Suspense, type ReactNode } from 'react';
+import Loading from './Loading';
+import ErrorSection from './ErrorSection';
+
+interface DataWrapperProps {
+  children: ReactNode;
+  loadingGuide?: string;
+  onRetry?: () => void;
+}
+
+function DataWrapper({ children, loadingGuide = '로딩 중...', onRetry }: DataWrapperProps) {
+  return (
+    <QueryErrorResetBoundary>
+      <ErrorBoundary
+        fallbackRender={({ error, resetErrorBoundary }) => (
```

Comment:
```suggestion
        fallbackRender={({ resetErrorBoundary }) => (
```

error는 사용되지 않으니 지워야할 것 같습니다~